oak/server/channel.h

```diff
@@ -77,13 +77,20 @@ class ChannelHalf {
   virtual void Write(std::unique_ptr<Message> msg) {
     // Fallback implementation that just drops the message.
   }
+
+  // Await blocks until there is a message avaialable to read on the channel.
+  virtual void Await() {}
 };
 
 // A concrete message channel, which holds an arbitrary number of queued messages.
-// Implements both send and receive ChannelHalf functions.
-class MessageChannel : public ChannelHalf {
+// Implements both send and receive ChannelHalf functions, though callers should wrap instances of
+// this class into the appropriate ChannelHalf implementation (|MessageChannelReadHalf| or
+// |MessageChannelWriteHalf|) to interact with it.
+// TODO: Clean up this interface, e.g. hide the Read and Write methods on MessageChannel and expose
+// methods to get access to the read and write halves instead.
+class MessageChannel final : ChannelHalf {
```

Comment:
Interesting &ndash; does `final` change the default inheritance access-specifier from `private` (the default for a deriving `class`) to `public`?  I guess it must.

Reply:
Why do we have final? Also, even if what @daviddrysdale is saying is true, we should have public there.

Reply:
Not intentional, added `public` back.

Reply:
`final` is to emphasize that this class should not be extended.